eclair-core/src/main/scala/fr/acinq/eclair/Setup.scala

```diff
@@ -181,12 +181,18 @@ class Setup(datadir: File,
         case true =>
           val host = config.getString("electrum.host")
           val port = config.getInt("electrum.port")
+          val address = InetSocketAddress.createUnresolved(host, port)
           val ssl = config.getString("electrum.ssl") match {
             case "off" => SSL.OFF
             case "loose" => SSL.LOOSE
             case _ => SSL.STRICT // strict mode is the default when we specify a custom electrum server, we don't want to be MITMed
           }
-          val address = InetSocketAddress.createUnresolved(host, port)
+
+          // turn SSL verification off for a Tor hidden service
+          if (address.getHostName().endsWith(".onion")) {
```

Comment:
Here is what I suggest:

```scala
val host = config.getString("electrum.host")
val port = config.getInt("electrum.port")
val address = InetSocketAddress.createUnresolved(host, port)
val ssl = if (address.getHostName.endsWith(".onion")) {
  // Tor already adds end-to-end encryption, so adding TLS on top doesn't add anything (apart from showing a
  // valid certificate which only makes sense for websites or paid APIs where users verify the certificate).
  SSL.OFF
} else {
  config.getString("electrum.ssl") match {
    case "off" => SSL.OFF
    case "loose" => SSL.LOOSE
    case _ => SSL.STRICT // strict mode is the default when we specify a custom electrum server, we don't want to be MITMed
  }
}
logger.info(s"override electrum default with server=$address ssl=$ssl")
Set(ElectrumServerAddress(address, ssl))
```

Reply:
Another suggestion (more compact and carries the intent a bit better I think):

```scala
val ssl = config.getString("electrum.ssl") match {
  case _ if address.getHostName.endsWith(".onion") => SSL.OFF // Tor already adds end-to-end encryption, adding TLS on top doesn't add anything
  case "off" => SSL.OFF
  case "loose" => SSL.LOOSE
  case _ => SSL.STRICT // strict mode is the default when we specify a custom electrum server, we don't want to be MITMed
}
```

Reply:
Once we settle on the code, @sstone said he could test it E2E before we merge.

Reply:
Thank you @pm47, great change! Just applied it https://github.com/ACINQ/eclair/pull/1278/commits/233bcdae3027e543e4b4f03afb8ac996b62c38dd.